frontend/src/components/dateSelector.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+
+export interface DateSelectorProps {
+  onChange: (value: Date) => void;
+  defaultDate: string | null | undefined;
+}
+
+const DateSelector: React.FC<DateSelectorProps> = ({
+  onChange,
+  defaultDate,
+}) => {
+  const [releaseDate, setReleaseDate] = useState<Date | null>(
+    defaultDate ? new Date(defaultDate) : null,
+  );
+  const [releaseTime, setReleaseTime] = useState<Date | null>(
+    defaultDate ? new Date(defaultDate) : null,
+  );
+
+  const [userChangedDate, setUserChangedDate] = useState<boolean>(false);
+  useEffect(() => {
+    if (userChangedDate && releaseDate && releaseTime) {
+      let newReleaseDateTime = releaseDate;
+      newReleaseDateTime.setHours(releaseTime.getUTCHours());
+      newReleaseDateTime.setMinutes(releaseTime.getUTCMinutes());
+      newReleaseDateTime.setSeconds(releaseTime.getUTCSeconds());
+      onChange(newReleaseDateTime);
+      console.log(newReleaseDateTime);
+    }
+  }, [releaseDate, releaseTime]);
+
+  //This is to convert the Date() object to a format that the input field accepts as defaultValue
+  const defaultDateAsDate =
+    defaultDate != null ? new Date(defaultDate) : undefined;
+  const defDate =
+    defaultDateAsDate != undefined
+      ? new Date(
+          defaultDateAsDate.getTime() -
+            defaultDateAsDate.getTimezoneOffset() * 60000,
```

Comment:
Will this work with summer time and winter time?

Reply:
```suggestion
          defaultDateAsDate.getTime() -
            defaultDateAsDate.getTimezoneOffset() * 60_000, // TimezoneOffset returns minutes -> Converting to milliseconds
```
Ah okay makes sense when I know that TimezoneOffset returns minutes